### PR TITLE
Detect outside meddling with the system clock and deal with it.

### LINKED ISSUE #717: ntpd-rs gets stuck when clock is adjusted while it is running
when the clock is moved (forward, did not test backward) while ntpd-rs is running, it will get stuck, printing 

```
2023-05-01T14:55:02.819620Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:17.925342Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:18.342352Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:19.160934Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:19.242889Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:34.270160Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:34.654405Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:35.546725Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:35.972728Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:50.608621Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
2023-05-01T14:55:51.119023Z  INFO ntp_proto::algorithm::kalman: No concensus cluster found
```

such an adjustment of the clock should not happen of course, but maybe we can do a better job of recognizing it?

### PR DESCRIPTION
**Pull request by davidv1992**
This patch ensures that any external change to the system clock resets the kalman algorithm, ensuring we at least keep operating when that happens even though it is less than ideal. Depends on #760 

Fixes #717

### CODE REVIEW

File: ntp-proto/src/algorithm/kalman/peer.rs [hunk 1]
```diff
@@ -421,7 +421,29 @@ impl PeerState {
...
+                // We check that the difference between the localtime and monotonic
+                // times of the measurement is in line with what would be expected
+                // from recent steering. This check needs to be done here since we
+                // need to revert back to the initial state.
+                if ((measurement.localtime - filter.last_measurement.localtime)
+                    - measurement
+                        .monotime
+                        .abs_diff(filter.last_measurement.monotime))
+                .abs()
+                    > algo_config.meddling_threshold
```
Comment (folkertdev) [inline ntp-proto/src/algorithm/kalman/peer.rs:434]: can this be broken up? e.g. give some good names to the intermediate results. This looks really weird and is hard to read

File: ntp-proto/src/algorithm/kalman/peer.rs [hunk 2]
```diff
@@ -569,7 +746,7 @@ mod tests {
                 transmit_timestamp: Default::default(),
                 receive_timestamp: Default::default(),
                 localtime: base + NtpDuration::from_seconds(1000.0),
-                monotime: basei,
+                monotime: basei + std::time::Duration::from_secs(1000),
```
Comment (folkertdev) [inline ntp-proto/src/algorithm/kalman/peer.rs:749]: why this change?

**Review by folkertdev — CHANGES_REQUESTED** [review]
I think something should be added to the changelog too for this change, because it adds a new config option.

**Comment by folkertdev** [comment]
also does this fix https://github.com/pendulum-project/ntpd-rs/issues/717 ?

**Comment by davidv1992** [comment]
Yes it does

File: ntp-proto/src/algorithm/kalman/peer.rs (hunk 1 above)
Comment (davidv1992) [reply ntp-proto/src/algorithm/kalman/peer.rs]: done.

File: ntp-proto/src/algorithm/kalman/peer.rs (hunk 2 above)
Comment (davidv1992) [reply ntp-proto/src/algorithm/kalman/peer.rs:751]: Because we now check that both clocks progress at the same rate, otherwise the update is rejected.

**Review by folkertdev — APPROVED** [review]
